Wikipedia/Code/WMFSaveButtonController.h

```diff
@@ -24,6 +24,7 @@
 /**
  *  Set to provide a source for logging saved pages
  */
-@property (weak, nonatomic) id<WMFAnalyticsLogging> analyticsSource;
+@property (weak, nonatomic) id<WMFAnalyticsContextProviding> analyticsContext;
```

Comment:
right, but cells are always reused w/in the same view, so the context won't change during its lifetime.